packages/patternfly-4/react-table/src/components/Table/Header.js

```diff
@@ -0,0 +1,98 @@
+import React from 'react';
+import { Header } from 'reactabular-table';
+import PropTypes from 'prop-types';
+import { TableContext } from './Table';
+import isEqual from 'lodash/isEqual';
+import { SortByDirection } from './SortColumn';
+import { calculateColumns } from './utils/headerUtils';
+
+const propTypes = {
+  className: PropTypes.string,
+  headerRows: PropTypes.array.isRequired,
+  sortBy: PropTypes.shape({
+    index: PropTypes.number,
+    direction: PropTypes.oneOf(Object.values(SortByDirection))
+  }),
+  onCollapse: PropTypes.func,
+  onSort: PropTypes.func,
+  onSelect: PropTypes.func,
+  selectLabeledBy: PropTypes.string,
+  actions: PropTypes.array
+};
+
+const defaultProps = {
+  onCollapse: null,
+  className: '',
+  sortBy: {},
+  onSort: null,
+  onSelect: null
+};
+
+class ContextHeader extends React.Component {
+  shouldComponentUpdate(nextProps) {
+    const { sortBy: nextSortBy, actions: nextAction } = nextProps;
+    const { sortBy, actions, onCollapse, headerRows } = this.props;
+    const updateCheckers = [
+      () => nextProps.onCollapse && onCollapse && !isEqual(nextProps.onCollapse, onCollapse),
```

Comment:
isEqual here seems unneeded. it is a single property, and a normal `===` check should be sufficient.